### PR TITLE
RavenDB-14609  Add Tombstones to Import Only (Removed from Export & M…

### PR DESCRIPTION
**Pull request by Danielle9897**
…igrate)

### CODE REVIEW

File: src/Raven.Studio/typescript/common/generalUtils.ts [hunk 1]
```diff
@@ -406,6 +406,16 @@ class genUtils {
         return inputNumber.toLocaleString(undefined, { minimumFractionDigits: n, maximumFractionDigits: n });
     };
 
+    /***  File Methods ***/
+    
+    static isFileOfType(fileName: string, possibleFileTypes: string[]): boolean {
+        const fileParts = _.split(fileName,".");
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/generalUtils.ts]: space after comma

File: src/Raven.Studio/typescript/common/generalUtils.ts [hunk 2]
```diff
@@ -406,6 +406,16 @@ class genUtils {
         return inputNumber.toLocaleString(undefined, { minimumFractionDigits: n, maximumFractionDigits: n });
     };
 
+    /***  File Methods ***/
+    
+    static isFileOfType(fileName: string, possibleFileTypes: string[]): boolean {
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/generalUtils.ts]: I would use even path here - as it should well for any path, not only fileName ;-)

Comment (ml054) [inline src/Raven.Studio/typescript/common/generalUtils.ts]: + I would be nice to have word extension in method name. 

Should we are checking if given path points to file with given extension.

File: src/Raven.Studio/typescript/common/generalUtils.ts [hunk 3]
```diff
@@ -406,6 +406,16 @@ class genUtils {
...
+        const fileParts = _.split(fileName,".");
+        const fileExtension = fileParts[fileParts.length-1];
+
+        return fileParts.length === 2 &&
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/generalUtils.ts]: this is wrong. What if file contains dots like: ravendb.net.backup? In general you should check if path ends with "." + extension.

**Review by ml054 — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/common/generalUtils.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/common/generalUtils.ts]: done

File: src/Raven.Studio/typescript/viewmodels/common/notificationCenter/detailViewer/operations/smugglerDatabaseDetails.ts [hunk 4]
```diff
@@ -106,8 +106,11 @@ class smugglerDatabaseDetails extends abstractOperationDetails {
                 result.push(this.mapToExportListItem("Compare Exchange", status.CompareExchange));
                 result.push(this.mapToExportListItem("Counters", status.Counters));
                 result.push(this.mapToExportListItem("Subscriptions", status.Subscriptions));
-                result.push(this.mapToExportListItem("Tombstones", status.Tombstones));
-                result.push(this.mapToExportListItem("Compare Exchange Tombstones", status.CompareExchangeTombstones));
+                
+                if (this.op.taskType() === "DatabaseImport") {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/common/notificationCenter/detailViewer/operations/smugglerDatabaseDetails.ts:110]: nice :)

File: src/Raven.Studio/wwwroot/App/views/database/tasks/importDatabaseFromFile.html [hunk 5]
```diff
@@ -2,8 +2,14 @@
 <div class="panel">
     <div class="panel-body">
         <form data-bind="submit: importDb">
-            <h3>Import data from a .ravendbdump file into the current database</h3>
-            <div class="bg-warning padding padding-xs margin-bottom"><small><i class="icon-warning"></i> Note: Importing will overwrite any existing documents and indexes.</small></div>
+            <div class="row">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/importDatabaseFromFile.html:5]: why do you need to introduce row and cols here?

File: src/Raven.Studio/wwwroot/App/views/database/tasks/importDatabaseFromFile.html [hunk 6]
```diff
@@ -16,6 +22,13 @@ <h3>Import data from a .ravendbdump file into the current database</h3>
                         </span>
                         <p class="help-block" data-bind="validationMessage: importedFileName"></p>
                     </div>
+                    <div class="text-warning bg-warning padding padding-xs margin-top flex-horizontal" data-bind="visible: isBackupFileType">
+                        <div class="flex-start"><small><i class="icon-warning"></i></small></div>
+                        <div>
+                            <div><small>The selected file is a <strong>RavenDB Backup file</strong>.</small></div>
+                            <div><small>Please use the <strong>'Restore'</strong> option (under Create New Database) in order to restore data from a RavenDB Backup file.</small></div>
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/importDatabaseFromFile.html]: maybe let's use strong w/o ' (single quote)

Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/database/tasks/importDatabaseFromFile.html]: done

File: src/Raven.Studio/wwwroot/App/views/database/tasks/importDatabaseFromFile.html (hunk 5 above)
Comment (ml054) [reply src/Raven.Studio/wwwroot/App/views/database/tasks/importDatabaseFromFile.html:5]: I can see to nicely match warning boxes. Let's leave it then.

File: src/Raven.Studio/typescript/common/generalUtils.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/common/generalUtils.ts]: done - using getFileExtension method

Comment (Danielle9897) [reply src/Raven.Studio/typescript/common/generalUtils.ts]: done - using getFileExtension method

File: src/Raven.Studio/typescript/common/generalUtils.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/common/generalUtils.ts]: done - using getFileExtension method

File: src/Raven.Studio/typescript/common/generalUtils.ts [hunk 7]
```diff
@@ -406,6 +406,14 @@ class genUtils {
         return inputNumber.toLocaleString(undefined, { minimumFractionDigits: n, maximumFractionDigits: n });
     };
 
+    /***  File Methods ***/
+    
+    static getFileExtension(filePath: string): string {
+        const fileParts = _.split(filePath, ".");
+        
+        return fileParts.length > 1 ? fileParts.pop() : null;
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/generalUtils.ts:414]: nice I forgot pop exists ;-)

**Review by ml054 — APPROVED** [review]

**Pull request merged**